crates/bevy_sprite/src/render/mod.rs

```diff
@@ -690,8 +694,7 @@ pub fn prepare_sprite_image_bind_groups(
                     });
 
                 batch_item_index = item_index;
-                batches.push((
-                    item.entity(),
+                current_batch = Some(batches.entry((*retained_view, item.main_entity())).insert(
```

Comment:
Do we need to use the whole `RetainedViewEntity` as a key, it seems like just using `retained_view.main_entity` would be enough to avoid collisions?

Reply:
I think it probably would be fine, especially because this is not yet retained and we are rebuilding this every frame, but we're moving towards using `RetainedViewEntity` everywhere for safety reasons. In my brief benching, using just the `MainEntity` didn't seem to have a notable performance impact.